webroot/src/components/Users/UserInvite/UserInvite.ts

```diff
@@ -0,0 +1,462 @@
+//
+//  UserInvite.ts
+//  CompactConnect
+//
+//  Created by InspiringApps on 11/11/2024.
+//
+
+import { Component, mixins, toNative } from 'vue-facing-decorator';
+import { reactive, computed, ComputedRef } from 'vue';
+import MixinForm from '@components/Forms/_mixins/form.mixin';
+import Card from '@components/Card/Card.vue';
+import MockPopulate from '@components/Forms/MockPopulate/MockPopulate.vue';
+import InputText from '@components/Forms/InputText/InputText.vue';
+import InputSelect from '@components/Forms/InputSelect/InputSelect.vue';
+import InputButton from '@components/Forms/InputButton/InputButton.vue';
+import InputSubmit from '@components/Forms/InputSubmit/InputSubmit.vue';
+import LoadingSpinner from '@components/LoadingSpinner/LoadingSpinner.vue';
+import { Compact } from '@models/Compact/Compact.model';
+import { State } from '@models/State/State.model';
+import {
+    StaffUser,
+    StaffUserSerializer,
+    CompactPermission,
+    StatePermission
+} from '@models/StaffUser/StaffUser.model';
+import { FormInput } from '@models/FormInput/FormInput.model';
+import Joi from 'joi';
+
+interface PermissionOption {
+    value: string | number;
+    name: string | ComputedRef<string>;
+    isDisabled?: boolean;
+}
+
+enum Permission {
+    NONE = 'none',
+    READ = 'read',
+    WRITE = 'write',
+    ADMIN = 'admin',
+}
+
+interface PermissionObject {
+    isRead?: boolean;
+    isWrite?: boolean;
+    isAdmin?: boolean;
+}
+
+@Component({
+    name: 'UserInvite',
+    components: {
+        Card,
+        MockPopulate,
+        InputText,
+        InputSelect,
+        InputButton,
+        InputSubmit,
+        LoadingSpinner,
+    },
+    emits: [ 'cancel', 'saved' ],
+})
+class UserInvite extends mixins(MixinForm) {
+    //
+    // Data
+    //
+    permissionStateInputs: Array<FormInput> = [];
+
+    //
+    // Lifecycle
+    //
+    created() {
+        this.shouldValuesIncludeDisabled = true;
+        this.initFormInputs();
+    }
+
+    //
+    // Computed
+    //
+    get userStore() {
+        return this.$store.state.user;
+    }
+
+    get currentUser(): StaffUser {
+        return this.userStore.model;
+    }
+
+    get currentCompact(): Compact | null {
+        return this.userStore.currentCompact;
+    }
+
+    get currentUserCompactPermission(): CompactPermission | null {
+        const currentPermissions = this.currentUser?.permissions;
+        const compactPermission = currentPermissions?.find((currentPermission: CompactPermission) =>
+            currentPermission.compact.type === this.currentCompact?.type) || null;
+
+        return compactPermission;
+    }
+
+    get isCurrentUserCompactAdmin(): boolean {
+        return this.getCompactPermission(this.currentUserCompactPermission) === Permission.ADMIN;
+    }
+
+    get isCurrentUserStateAdminAny(): boolean {
```

Comment:
What's "Any" supposed to mean? Might `isStateAdmin` be clearer? (I'm mirroring `isAnyTypeOfAdmin` which doesn't explicitly name the current user, which I think is implied/a fair assumption?)

I guess separately, I wonder if this might make more sense to be on `currentUser` directly? Without digging into it, this code feels familiar in that I've seen similar logic in prior PRs (I may be wrong). Fine to leave as-is for this PR, but I wonder if we might consider refactoring in the future.

Reply:
The "Any" is just trying to disambiguate from any methods that might resemble `isCurrentUserStateAdmin(state)`. 

Totally agree about moving & factoring some of these common things to the user instance in a future PR. There was some ambiguity and nuance surrounding these user management features. I've got a call w/ @isabeleliassen later this week to walk through all of the user management now that it's roughly feature-complete (once this is merged). 

I've also create #342 to specifically track any code factoring once that walkthrough and any resulting change requests are complete.